program-analysis/echidna/basic/working-with-eth.md

```diff
@@ -0,0 +1,119 @@
+# Using ether during a fuzzing campaign
+
+**Table of contents:**
+
+- [Using ether during a fuzzing campaign](#using-ether-during-a-fuzzing-campaign)
+  - [Introduction](#introduction)
+  - [Controlling the amount of ether in payable functions](#controlling-the-amount-of-ether-in-payable-functions)
+  - [Controlling the amount of ether in contracts](#controlling-the-amount-of-ether-in-contracts)
+  - [Summary: Working with ether](#summary-working-with-ether)
+
+## Introduction
+
+We will see how to use ETH during the fuzzing fuzzed. The following smart contract will be used as example:
+
+```solidity
+contract C {
+    function pay() public payable {
+        require(msg.value == 12000);
+    }
+
+    function echidna_has_some_value() public returns (bool) {
+        return (address(this).balance != 12000);
+    }
+}
+```
+
+This code forces Echidna to send a particular amount of eth as value in the `pay` function.
+Echidna will do this for each payable function in the target function (or any contract if `allContracts` is enabled):
+
+```
+$ echidna balanceSender.sol
+...
+echidna_has_some_value: failed!💥
+  Call sequence:
+    pay() Value: 0x2ee0
+```
+
+Echidna will show the value amount in hexadecimal.
+
+## Controlling the amount of ether in payable functions
+
+The amount of ether to send in each payable function will be randomly selected, but with a maximum value determined by the `maxValue` value
+with a default of 100 eth per transaction:
+
+```yaml
+maxValue: 100000000000000000000
+```
+
+This means that each transaction will contain, at most, 100 eth in value. However, there is no maximum that will be used in total.
+The maximum amount to receive will be determined by the number of transactions. If you are using 100 transactions (`--seq-len 100`),
+then the maximum amount of ether in transactions will be 100 \* 100 eth.
```

Comment:
- `then the maximum amount of ether in transactions will be 100 \* 100 eth` -> this might seem like there always be this maximum amount. Maybe `then the total amount of ether used for all the transactions will be between 0 and 100 * 100 ethers`

Reply:
Fixed